File: gestionstock/src/main/java/obrasmart/gestionstock/service/CategoriaRepuestoService.java
package obrasmart.gestionstock.service;

import obrasmart.gestionstock.dto.CategoriaRepuestoDto;

import java.util.List;

public interface CategoriaRepuestoService {

        List<CategoriaRepuestoDto> listar();

        CategoriaRepuestoDto buscar(Long id);

        CategoriaRepuestoDto crear(CategoriaRepuestoDto dto);

        CategoriaRepuestoDto actualizar(Long id, CategoriaRepuestoDto dto);

        void eliminar(Long id);

}

File: frontend-obrasmart2/src/app/service/categoria-repuesto.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';

export interface CategoriaRepuestoDTO {
  id?: number;
  nombre: string;
  descripcion?: string;
}

@Injectable({ providedIn: 'root' })
export class CategoriaRepuestoService {
  private apiUrl = `${environment.apiUrl}/categorias`;

  constructor(private http: HttpClient) {}

  listar(): Observable<CategoriaRepuestoDTO[]> {
    return this.http.get<CategoriaRepuestoDTO[]>(this.apiUrl);
  }

  crear(dto: CategoriaRepuestoDTO): Observable<CategoriaRepuestoDTO> {
    return this.http.post<CategoriaRepuestoDTO>(this.apiUrl, dto);
  }

  eliminar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
